src/storage-client/src/sink.rs

```diff
@@ -202,13 +229,19 @@ where
 
     let retention_ms_str = retention.duration.map(|d| d.to_string());
     let retention_bytes_str = retention.bytes.map(|s| s.to_string());
-    if let Some(ref retention_ms) = retention_ms_str {
+    if let Some(retention_ms) = &retention_ms_str {
         kafka_topic = kafka_topic.set("retention.ms", retention_ms);
     }
-    if let Some(ref retention_bytes) = retention_bytes_str {
+    if let Some(retention_bytes) = &retention_bytes_str {
         kafka_topic = kafka_topic.set("retention.bytes", retention_bytes);
     }
 
+    let mut cleanup_policy = "delete";
+    if compaction {
+        cleanup_policy = "delete,compact";
```

Comment:
💭 This is correct, but I'd also be tempted to ban mixing these... I've never seen a case where it's actually what you want. (vs. https://kafka.apache.org/documentation/#topicconfigs_max.compaction.lag.ms which has some advantages.)

Reply:
Fair point! I was trying to keep the utility function as general purpose as possible, but happy to make its signature more prescriptive and force the caller to choose between retention and compaction. Will push up a revision to handle that.

Reply:
Done.